src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box } from '@chakra-ui/react'
+import { TRPCClientError } from '@trpc/client'
+import { TRPC_ERROR_CODE_KEY } from '@trpc/server/rpc'
+import { Component } from 'react'
+import { ErrorBoundaryProps, ErrorBoundaryState } from './ErrorBoundary.types'
+import { TRPCWithErrorCodeSchema } from '../../utils/error'
+import { UnexpectedErrorCard } from './UnexpectedErrorCard'
+
+class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props)
+
+    // Define a state variable to track whether is an error or not
+    this.state = { hasError: false }
+  }
+  static getDerivedStateFromError(error: unknown) {
+    // Update state so the next render will show the fallback UI
+    return { hasError: true, error }
+  }
+  componentDidCatch(error: Error) {
+    // You can use your own error logging service here
+    console.log({ error })
+  }
+
+  render() {
+    const { children, fallback } = this.props
+    const error = this.state.error
+
+    // Check if the error is thrown
+    if (!this.state.hasError) {
+      return children
+    }
+
+    if (fallback !== undefined) {
+      return fallback
+    }
+
+    if (error instanceof TRPCClientError) {
+      const res = TRPCWithErrorCodeSchema.safeParse(error)
+
+      if (!res.success) return <UnexpectedErrorCard />
+
+      // The choice to not redirect via next's router was intentional to handle ErrorBoundary for the app root
+      // Using next's router.push('/sign-in') will not render the SignIn component as it won't be mounted in the app root as the ErrorBoundary fallback component will be rendered instead
+      // Using vanilla location redirecting will prompt a full page reload of /sign-in page, which will never trigger the root ErrorBoundary, thus rendering the full component correctly
+      if (res.data === 'UNAUTHORIZED') {
+        window.location.href = 'sign-in'
+        return
+      }
```

Comment:
@karrui @dvrylc check out this statement for the root level ErrorBoundary and lmk what you think. We can walk through tomorrow if docs arent clear.